### ISSUE TITLE
SPFX Build time is too long. 🐌

### ISSUE THREAD

**Issue by nayots** [SharePoint/sp-dev-docs#4797]
#### Category
- [x] Question
- [ ] Typo
- [ ] Bug
- [ ] Additional article idea

#### Expected or Desired Behavior

When you are serving a spfx project , with possibly multiple webparts and extensions, the build time should be close to the build time of "create-react-app".

#### Observed Behavior

It takes multiple minutes for the serve to build the solution and serve it with reload.

#### Steps to Reproduce

Create a solution with multiple webparts with react. Build it.

TLDR: build time is too long, and there is no hot reload.

**Comment by CPritch** [comment]
In the generated gulpfile it always runs the Webpack Bundle Analyzer which takes a long time. We have modified this to only run when the production flag is enabled.

Change this part of your `gulpfile.js` from:
```js
build.configureWebpack.mergeConfig({

	additionalConfiguration: (generatedConfiguration) => {
		const lastDirName = path.basename(__dirname);
		const dropPath = path.join(__dirname, 'temp', 'stats');
		generatedConfiguration.plugins.push(new bundleAnalyzer.BundleAnalyzerPlugin({
			openAnalyzer: false,
			analyzerMode: 'static',
			reportFilename: path.join(dropPath, `${lastDirName}.stats.html`),
			generateStatsFile: true,
			statsFilename: path.join(dropPath, `${lastDirName}.stats.json`),
			logLevel: 'error'
		}));

		return generatedConfiguration;
	}
});
```

To this:
```js
build.configureWebpack.mergeConfig({

	additionalConfiguration: (generatedConfiguration) => {
		const lastDirName = path.basename(__dirname);
		const dropPath = path.join(__dirname, 'temp', 'stats');
                // Only analyze bundle on production builds
		if (process.argv.indexOf('--ship') > -1)
			generatedConfiguration.plugins.push(new bundleAnalyzer.BundleAnalyzerPlugin({
				openAnalyzer: false,
				analyzerMode: 'static',
				reportFilename: path.join(dropPath, `${lastDirName}.stats.html`),
				generateStatsFile: true,
				statsFilename: path.join(dropPath, `${lastDirName}.stats.json`),
				logLevel: 'error'
			}));

		return generatedConfiguration;
	}
});
```

Theres not too many ways of improving bundle speeds that I know of. The key thing is to move to test driven development as you don't have to wait for a bundle to complete before knowing the result of changes to the codebase.

**Comment by andrewconnell** [comment]
Respectfully, this isn't a question/bug or actionable... it's just venting. "It should be as fast as _this_ unrelated project type" is arbitrary.

WRT build times, there have been some good blog posts on the subject, such as:
https://spblog.net/post/2019/09/18/spfx-overclockers-or-how-to-significantly-improve-your-sharepoint-framework-build-performance
